assets/src/components/form/model/base/model-select.js

```diff
@@ -130,6 +159,71 @@ export class ModelSelect extends Component {
 		return null;
 	}
 
+	/**
+	 * Helper method for determining whether state should be updated.
+	 * Criteria is whether the keys from the incoming entities are equivalent
+	 * (equivalency means in same order as well)
+	 *
+	 * @param {Object} prevProps
+	 * @param {Object} nextProps
+	 * @return {boolean} True means state should be updated.
+	 */
+	static shouldUpdateState( prevProps, nextProps ) {
```

Comment:
So this method cut down on some unnecessary renders by checking the incoming props to see whether another render is necessary. When state is updated, that means the child components will get updated (in case the `react-select` component).

I think I've covered in here all the specific conditions that should trigger a re-render of the react-select component but its possible we may uncover some other conditions as we start using these more.  I think its better we err on the side of caution initially to avoid unnecessary re-renders.